user/showthread.php

```diff
@@ -6,6 +6,14 @@
 require_once("message.inc");
 require_once("page-yatt.inc.php");
 
+if(isset($forum['option']['LoginToRead']) and $forum['option']['LoginToRead']) {
+  $user->req();
+  if ($user->status != 'Active') {
+    echo "Your account isn't validated\n";
+    exit;
+  }
+}
+
```

Comment:
I see this code block copy/pasted between a bunch of different files; maybe a single common function would make sense.

Reply:
Here is another variant in showtracking.php
```
if (!$user->valid()) header("Location: /login.phtml?url=$url");
```

Reply:
Either would work, what I submitted has been implemented in a production environment without issue.

Reply:
Will merge this for now, and will open a new ticket for making common code.